### PR TITLE
Fix catching k8s API errors

### PR DESCRIPTION
**Pull request by nprokopic**
Towards https://github.com/giantswarm/giantswarm/issues/15156

What's in the box:
- Fixes issue where API conflict error happens in `azuremachinepooolconditions` handler when updating `AzureMachinePool` CR.
- Fixes issue where, during cluster upgrade, workload cluster API call fails due to API being unavailable.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,8 +15,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Ensure that availability zones are kept unchanged during migration from 12.x to 13.x.
 - Don't set `MachinePool.Status.InfrastructureReady` in `nodestatus` handler.
-- Update `giantswarm/conditions-handler` to `v0.1.2`.
```
Comment (whites11) [inline CHANGELOG.md:18]: Why did you remove this?

**Review by whites11 — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (nprokopic) [reply CHANGELOG.md:18]: There is this few lines above:

```
- Bump `conditions-handler` to v0.2.0 to get `MachinePool` `ReplicasReady` condition.
```
I updated PR description.

Comment (nprokopic) [reply CHANGELOG.md:18]: Forgot to update it before :)

**Review by tuommaki — APPROVED** [review]
LGTM
